scripts/download-resources.mjs

```diff
@@ -101,54 +111,164 @@ export async function download(url, destPath, expectedSHA = '', overwrite = fals
 }
 
-export default async function main() {
-  const resourcesDir = path.join(process.cwd(), 'resources', os.platform());
-  const binDir = path.join(resourcesDir, 'bin');
-  /** The platform string, as used by golang / Kubernetes. */
-  const kubePlatform = {
-    darwin: 'darwin',
-    linux:  'linux',
-    win32:  'windows',
-  }[os.platform()];
+/**
+ * Find the home directory, in a way that is compatible with
+ * kuberlr
+ */
+async function findHome() {
+  const tryAccess = async(path) => {
+    try {
+      await fs.promises.access(path);
+
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  const osHomeDir = os.homedir();
 
+  if (osHomeDir && await tryAccess(osHomeDir)) {
+    return osHomeDir;
+  }
+  if (process.env.HOME && await tryAccess(process.env.HOME)) {
+    return process.env.HOME;
+  }
+  if (onWindows) {
+    if (process.env.USERPROFILE && await tryAccess(process.env.USERPROFILE)) {
+      return process.env.USERPROFILE;
+    }
+    if (process.env.HOMEDRIVE && process.env.HOMEPATH) {
+      const homePath = path.join(process.env.HOMEDRIVE, process.env.HOMEPATH);
+
+      if (await tryAccess(homePath)) {
+        return homePath;
+      }
+    }
+  }
+
+  return null;
+}
+
+async function downloadKuberlr(kuberlrBaseURL, finalKuberlrSHA, kuberlrPlatformDir, onWindows) {
+  if (onWindows) {
+    return await downloadZip(`${ kuberlrBaseURL }.zip`, finalKuberlrSHA, 'kuberlr', kuberlrPlatformDir);
+  }
+
+  return await downloadTarGZ(`${ kuberlrBaseURL }.tar.gz`, finalKuberlrSHA, 'kuberlr', kuberlrPlatformDir);
+}
+
+export default async function main() {
   fs.mkdirSync(binDir, { recursive: true });
 
-  // Download Kubectl
-  const kubeVersion = '1.20.7';
-  const kubectlURL = `https://dl.k8s.io/v${ kubeVersion }/bin/${ kubePlatform }/amd64/${ exeName('kubectl') }`;
+  const kuberlrVersion = '0.3.2';
+  const kuberlrBase = `https://github.com/flavio/kuberlr/releases/download/v${ kuberlrVersion }`;
+  const kuberlrBaseURL = `${ kuberlrBase }/kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`;
+  const kuberlrPlatformDir = `kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`;
+  const allKuberlrSHAs = await getResource(`${ kuberlrBase }/checksums.txt`);
+  const kuberlrSHA = allKuberlrSHAs.split(/\r?\n/).filter(line => line.includes(`kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`));
+
+  switch (kuberlrSHA.length) {
+  case 0:
+    throw new Error(`Couldn't find a matching SHA for [kuberlr_${ kuberlrVersion }_${ kubePlatform }-amd64] in [${ allKuberlrSHAs }]`);
+  case 1:
+    break;
+  default:
+    throw new Error(`Matched ${ kuberlrSHA.length } hits, not exactly 1, for platform ${ kubePlatform } in [${ allKuberlrSHAs }]`);
+  }
+  const finalKuberlrSHA = kuberlrSHA[0].split(/\s+/, 1)[0];
+  const kuberlrPath = await downloadKuberlr(kuberlrBaseURL, finalKuberlrSHA, kuberlrPlatformDir, onWindows);
+
+  // Download Kubectl into kuberlr's directory of versioned kubectl's
```

Comment:
Hmm, _not_ part of this PR, but I wonder if we should file a RFE on kuberlr to be able to manually trigger downloads of versions of kubectl?

Reply:
Done in https://github.com/flavio/kuberlr/issues/11